File: lena-config/src/main/java/org/riversoforion/lena/config/resolvers/SystemPropertiesValueResolver.java
/*
 * Copyright (c) 2024. Eric McIntyre / Rivers of Orion
 */
package org.riversoforion.lena.config.resolvers;

import org.riversoforion.lena.config.ValueResolver;

import java.util.Optional;

public class SystemPropertiesValueResolver implements ValueResolver {

    @Override
    public Optional<String> resolveValue(String name) {

        return Optional.ofNullable(System.getProperty(name));
    }
}

File: lena-config/src/jvmMain/kotlin/org/riversoforion/lena/config/resolvers/SystemPropertiesValueResolver.kt
/*
 * Copyright (c) 2024-2026. Eric McIntyre / Rivers of Orion
 */
package org.riversoforion.lena.config.resolvers

import org.riversoforion.lena.config.ValueResolver

/** Looks up values from JVM system properties. JVM-only — no native equivalent. */
public class SystemPropertiesValueResolver : ValueResolver {
    override fun resolveValue(name: String): String? = System.getProperty(name)
}
